### PR TITLE
DRILL-6287: apache-release profile should be disabled by default

### PR DESCRIPTION
**Pull request by vrozov**
@parthchandra @dvjyothsna Please review

### CODE REVIEW

**Comment by vrozov** [comment]
@parthchandra Please review

**Comment by vrozov** [comment]
There are two issues with enabling `apache-release` by default:
-  it triggers creating source `apache-drill-...-src.tar.gz` and `apache-drill-...-src.zip` archives.
- maven build for any sub-module fails.

The change disables activation of the `apache-release` profile based on JDK version and requires explicit activation during the Apache release process.

JDK 1.7 is not supported. See DRILL-1491 and #1143.

**Comment by parthchandra** [comment]
Sorry, Maven not being a strong point, I didn't understand initially what I was looking at.

+1

File: pom.xml [hunk 1]
```diff
@@ -66,6 +66,7 @@
     <excludedGroups />
     <memoryMb>4096</memoryMb>
     <directMemoryMb>4096</directMemoryMb>
+    <additionalparam>-Xdoclint:none</additionalparam>
```
Comment (vdiravka) [inline pom.xml]: Do we need a task (new Jira) for refactoring the Drill java docs and moving onto Java 8 doclint?

Comment (vrozov) [reply pom.xml]: @vdiravka Please see DRILL-4547.

Comment (vdiravka) [reply pom.xml]: Thanks

**Pull request closed without merging**